### ISSUE TITLE
Fix S1449: Rule should not report for objects when ToUpper is inside expression

### ISSUE THREAD

**Issue by valhristov** [SonarSource/sonar-csharp#1841]
### Description

S1449 incorrectly reports for the `ToUpper` invocation below. If the code is changed as per the recommendation, the query will cease working.

Note: the issue was originally reported for VB.NET, but it should also apply for C#

```
Dim londonCustomers = From customer in Customers _
	Where customer.City = “London” && _
	customer.code.ToUpper().Trim() == “Somthing”
Select customer
```

Original community thread:
https://community.sonarsource.com/t/c-rule-culture-should-be-specified-for-string-operations/2364

**Comment by paulhickman-a365** [comment]
This would fail with LINQ to many other things - not just entities.

It would be safer to suspend the rule anywhere that the ToUpper() is part of an Expression.

**Comment by valhristov** [comment]
Makes sense, @paulhickman-a365.
